http://bugs.gentoo.org/201646
http://sources.redhat.com/bugzilla/show_bug.cgi?id=5441

2007-12-07  Ulrich Drepper  <drepper@redhat.com>

	[BZ #5441]
	* stdio-common/vfscanf.c (_IO_vfwscanf): Don't free ptrs_to_free
	structure, it's allocated with alloca.

---
 stdio-common/vfscanf.c |    1 -
 1 file changed, 1 deletion(-)

Index: glibc-2.7/stdio-common/vfscanf.c
===================================================================
--- glibc-2.7.orig/stdio-common/vfscanf.c
+++ glibc-2.7/stdio-common/vfscanf.c
@@ -2843,7 +2843,6 @@ _IO_vfscanf_internal (_IO_FILE *s, const
 		  *p->ptrs[cnt] = NULL;
 		}
 	      p = p->next;
-	      free (ptrs_to_free);
 	      ptrs_to_free = p;
 	    }
 	}
